.travis.yml

```diff
@@ -105,6 +105,7 @@ before_install:
         export JULIA_TEST_MAXRSS_MB=600;
         TESTSTORUN="all --skip linalg/triangular subarray"; fi # TODO: re enable these if possible without timing out
     - echo "override JULIA_CPU_TARGET=generic;native" >> Make.user
+    - echo "override USECCACHE=1" >> Make.user
```

Comment:
Don't use `USECCACHE=1` that option is quite broken.
```
CC=/usr/lib/ccache/bin/gcc
CXX=/usr/lib/ccache/bin/g++
```

works better. cc @maleadt

Reply:
I used it here because we're using it on CircleCI. It appears we even work around the `-Qunused-arguments` thing with Clang in Make.inc. The logic looks sound; what's broken about it?

Reply:
with `USECCACHE`, the `CC` arg becomes `ccache gcc`, which in the case of cmake is implemented by setting eg. `CMAKE_C_COMPILER` to `ccache` and `CMAKE_C_COMPILER_ARG1` to `gcc`. IIRC, this breaks other invocations of the toolchain, eg. with `ar`. Can't seem to find the relevant bug reports right now, but either way there's a much more robust way to invoke `ccache`: by using a symlink named eg. `gcc` pointing to the `ccache` binary. This avoids aforementioned difficulties in configuring the build.

Reply:
I have not witnessed any problems with setting `USECCACHE=1` on Linux builds.  I do this for all builds on the buildbot, from scratch or otherwise.